src/Handler/Command.hs

```diff
@@ -64,9 +70,14 @@ getCommandR token = do
             $(widgetFile "command")
 
 deleteCommandR :: Token -> Handler ()
-deleteCommandR token = unsafeRunStorage $ do
-    del token
-    del $ History token
+deleteCommandR token = do
+    archived <- unsafeRunStorage $ do
```

Comment:
Is the `Handler` monad an instance of `MonadError`? You could use `throwError`.

https://hackage.haskell.org/package/mtl-2.2.1/docs/Control-Monad-Except.html